src/node/commit.rs

```diff
@@ -64,16 +64,6 @@ pub struct CommitData<J> {
     phantom: PhantomData<J>,
 }
 
-impl<J> PartialEq for CommitData<J> {
-    // Two nodes are equal if they both have IMRs and those IMRs are equal.
-    fn eq(&self, other: &Self) -> bool {
-        self.imr
-            .zip(other.imr)
-            .map(|(a, b)| a == b)
-            .unwrap_or(false)
-    }
-}
```

Comment:
d7021c2447a3304a818c9c2848750c2c227ca0e9: I think this is good since we might want to use `CommitNode` in hash maps and the likes.

I don't like the recursive equality check on `FinalArrow` but we can fix that later.

Reply:
My bad; I should have looked a level deeper :) In this case, we are good.